extensions/HuggingFaceTransformers/python/src/aiconfig_extension_hugging_face_transformers/text_2_image.py

```diff
@@ -0,0 +1,341 @@
+        return completion_data
+
+    async def run_inference(
+        self, prompt: Prompt, aiconfig : "AIConfigRuntime", options : InferenceOptions, parameters: Dict[str, Any]
+    ) -> List[Output]:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            InferenceResponse: The response from the model.
+        """
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [pipeline_creation_data, _] = refine_pipeline_creation_params(model_settings)
+        if not pipeline_creation_data.get("requires_safety_checker", True):
+            pipeline_creation_data["safety_checker"] = None
+
+        pipeline_building_disclaimer_message = """
+Building the pipeline... This can take a long time if you haven't created one before 
+on this machine. Please be patient!
+
+If this seems to be taking too long, you can change some of the pipeline generation
+params. See more details here:
+https://huggingface.co/docs/diffusers/using-diffusers/loading
+"""
+        print(pipeline_building_disclaimer_message)
+
+        model_name : str = aiconfig.get_model_name(prompt)
+        # TODO (rossdanlm): Figure out a way to save model and re-use checkpoint
+        # Otherwise right now a lot of these models are taking 5 mins to load with 50
+        # num_inference_steps (default value). See here for more details:
+        # https://huggingface.co/docs/diffusers/using-diffusers/loading#checkpoint-variants
+        if isinstance(model_name, str) and model_name not in self.generators:
+            self.generators[model_name] = AutoPipelineForText2Image.from_pretrained(
+                pretrained_model_or_path=model_name,
+                safety_checker=None,
+                **pipeline_creation_data
+            )
+        generator = self.generators[model_name]
+
+        disclaimer_long_response_print_message = """\n
+Calling image generation. This can take a long time, (up to SEVERAL MINUTES depending
+on the model), please hold on...
+
+If this seems to be taking too long, you can change some of the params. For example, 
+you can set the `num_inference_steps` to a lower value at the cost of reduced image
+quality. See full list of params here:
+https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/text2img#diffusers.StableDiffusionPipeline.__call__
+
+If that doesn't work, you can also try less computationally intensive models. 
+        """
+        print(disclaimer_long_response_print_message)
+
+        completion_data = await self.deserialize(prompt, aiconfig, options, parameters)
+        response : Union[StableDiffusionPipelineOutput, StableDiffusionXLPipelineOutput] = generator(**completion_data)
```

Comment:
Why are the types for response set to StableDiffusion? Do all diffusion models use the same types?

Reply:
There is one more common one called Kandinsky 2.2. I made an issue for that in https://github.com/lastmile-ai/aiconfig/issues/471 to verify. I think as long as we support the most popular 3 as stated by their wiki we should be ok, just put in a disclaimer in the docstrings